src/vs/workbench/contrib/debug/browser/debugEditorActions.ts

```diff
@@ -41,17 +50,50 @@ class ToggleBreakpointAction extends EditorAction2 {
 				primary: KeyCode.F9,
 				weight: KeybindingWeight.EditorContrib
 			},
-			menu: {
-				when: CONTEXT_DEBUGGERS_AVAILABLE,
-				id: MenuId.MenubarDebugMenu,
-				group: '4_new_breakpoint',
-				order: 1
-			}
+			menu: [
+				{
+					when: CONTEXT_DEBUGGERS_AVAILABLE,
+					id: MenuId.MenubarDebugMenu,
+					group: '4_new_breakpoint',
+					order: 1
+				},
+				{
+					id: MenuId.DebugDisassemblyViewContext,
+					group: 'debug',
+					order: 1000,
+				}
+			],
 		});
 	}
 
-	async runEditorCommand(accessor: ServicesAccessor, editor: ICodeEditor, ...args: any[]): Promise<void> {
-		// TODO: add disassembly F9
+	async run(accessor: ServicesAccessor, ...args: any[]) {
+		// Find the editor with text focus or active
+		const codeEditorService = accessor.get(ICodeEditorService);
+		const editor = codeEditorService.getFocusedCodeEditor() || codeEditorService.getActiveCodeEditor();
+		if (!editor) {
+			// maybe it is disassemblyView.
+			const editorService = accessor.get(IEditorService);
+			if (editorService.activeEditor?.typeId === DisassemblyViewInput.ID) {
+				const view = editorService.activeEditorPane as DisassemblyView;
+				const selection = view?.getSelection();
```

Comment:
These work on the selected row, not the one I right-clicked. You probably want the focused row, not selected.